tests/unit/typeTagParser.test.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  StructTag,
+  TypeTagStruct,
+  TypeTagU8,
+  AccountAddress,
+  TypeTagU16,
+  TypeTagU32,
+  TypeTagU128,
+  TypeTagU64,
+  TypeTagU256,
+  TypeTagBool,
+  TypeTagAddress,
+  TypeTagVector,
+  stringStructTag,
+  objectStructTag,
+  optionStructTag,
+  TypeTag,
+} from "../../src";
+import { Identifier } from "../../src/transactions/instances";
+import { parseTypeTag } from "../../src/transactions/typeTag/typeTagParser";
+
+const MODULE_NAME = new Identifier("tag");
+const STRUCT_NAME = new Identifier("Tag");
+
+const structTagType = (typeTags?: Array<TypeTag>) =>
+  new TypeTagStruct(new StructTag(AccountAddress.ONE, MODULE_NAME, STRUCT_NAME, typeTags ?? []));
+
+describe("TypeTagParser", () => {
+  test("invalid types", () => {
+    // Missing 8
+    expect(() => parseTypeTag("8")).toThrow();
```

Comment:
can also add the expected message it throws to the test
```
toThrow("Missing 8")
```

Reply:
The messages need to be standardized a bit (which is why I didn't add any of them).

Will take a look at adding them.